src/Components/WebAssembly/WebAssembly/src/Services/WebAssemblyNavigationManager.cs

```diff
@@ -17,14 +22,17 @@ internal class WebAssemblyNavigationManager : NavigationManager
         /// </summary>
         public static WebAssemblyNavigationManager Instance { get; set; }
 
+        public WebAssemblyLazyLoadDefinition LazyLoadDefinition { get; set;  }
+
         public WebAssemblyNavigationManager(string baseUri, string uri)
         {
             Initialize(baseUri, uri);
         }
 
-        public void SetLocation(string uri, bool isInterceptedLink)
+        public async Task SetLocation(string uri, bool isInterceptedLink)
         {
             Uri = uri;
+            await BeforeLocationChangeAsync();
             NotifyLocationChanged(isInterceptedLink);
```

Comment:
This now async and is not being awaited, isn't it?

Reply:
The navigation manager shouldn't be in charge of loading the assemblies, this should be done by the Router